File: src/main/java/com/redbox/domain/donation/dto/Top5DonorWrapper.java
package com.redbox.domain.donation.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lombok.Getter;

import java.util.List;

@Getter
@JsonDeserialize
public class Top5DonorWrapper {

    private List<Top5DonorResponse> donors;

    @JsonCreator
    public Top5DonorWrapper() {
    }

    public Top5DonorWrapper(List<Top5DonorResponse> donors) {
        this.donors = donors;
    }
}

File: migration/src/main/kotlin/com/redbox/domain/donation/dto/Top5DonorWrapper.kt
package com.redbox.domain.donation.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.databind.annotation.JsonDeserialize

@JsonDeserialize
@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
data class Top5DonorWrapper(
    val donors: List<Top5DonorResponse>
) {
    @JsonCreator
    constructor() : this(emptyList())
}